src/lsp/sql_ast/sql_ast.ml

```diff
@@ -0,0 +1,585 @@
+(*                                                                        *)
+(* All rights reserved.                                                   *)
+(* This source code is licensed under the GNU Affero General Public       *)
+(* License version 3 found in the LICENSE.md file in the root directory   *)
+(* of this source tree.                                                   *)
+(*                                                                        *)
+(**************************************************************************)
+open Cobol_common
+(**************************************************************************)
+(*                              AST                                       *)
+(**************************************************************************)
+type sqlVarToken = string with_loc 
+[@@deriving ord]
+type cobolVarId = string with_loc
+[@@deriving ord]
+
+type cobol_var =
+| NotNull of cobolVarId
+| NullIndicator of cobolVarId * cobolVarId
+[@@deriving ord]
+
+type variable = 
+  | SqlVar of sqlVarToken
+  | CobolVar of cobol_var
+  [@@deriving ord]
+  
+type literal =
+  | LiteralVar of variable
+  | LiteralNum of string with_loc
+  | LiteralStr of string with_loc
+  | LiteralDot of string with_loc list
+  [@@deriving ord]
+
+type sql_token = 
+  | SqlInstr of string
+  | SqlVarToken of variable
+  | SqlLit of literal
+  | SqlQuery of sql_query
+  | SqlEquality of sql_equal (*TODO: remove*)
+  | SqlSearchCondition of search_condition (*TODO: remove*)
+
+and sql_instruction = sql_token list
+
+and complex_literal =
+  | SqlCompLit of literal
+  | SqlCompAs of literal * sqlVarToken (*ex: SMT AS INT*)
+  | SqlCompFun of sqlVarToken * sql_op list
+  | SqlCompStar
+
+and esql_instuction =
+  | At of variable * esql_instuction 
+  | Sql of sql_instruction
+  | Begin 
+  | BeginDeclare
+  | Exeption of begin_end_stm
+  | EndDeclare
+  | StartTransaction
+  | Whenever of whenever_condition * whenever_continuation
+  | Include of sqlVarToken
+  | Connect of connect_syntax
+  | Rollback of rb_work_or_tran option * rb_args option
+  | Commit of rb_work_or_tran option * bool
+  | Savepoint of variable
+  | SelectInto of (cobol_var) list * sql_select * sql_select_option list (*select and option_select*)
+  | DeclareTable of literal * ((sqlVarToken * sql_type) list)
+  | DeclareCursor of cursor
+  | Prepare of sqlVarToken * sql_instruction 
+  | ExecuteImmediate of sql_instruction
+  | ExecuteIntoUsing of sqlVarToken * (cobol_var list) option * (cobol_var list) option
+  | Disconnect of variable option (*db_id*)
+  | DisconnectAll
+  | Open of sqlVarToken * (cobol_var list) option  (*cursor name*)
+  | Close of sqlVarToken (*cursor name*)
+  | Fetch of sql_instruction * (cobol_var) list
+  | Insert of table * (value list)
+  | Delete of sql_instruction
+  | Update of sqlVarToken * sql_update * (update_arg option)
+  | Ignore of sql_instruction
+
+and begin_end_stm = esql_instuction * exeption list
```

Comment:
What do `begin/end` denote in SQL? I think it's better to properly name the type, and use records instead of tuples like this. `type try_block(?) = { try_instruction: esql_instruction with_loc; try_exceptions: exception list }`.

Note it's `exception` instead of `exeption`.